sdk/compute/Azure.ResourceManager.Compute/tests/Scenario/GalleryImageVersionContainerTests.cs

```diff
@@ -0,0 +1,216 @@
+            var lroImageVersion = await container.CreateOrUpdateAsync(GalleryImageVersionName, BasicGalleryImageVersionData);
+            GalleryImageVersion imageVersion = lroImageVersion.Value;
+            Assert.IsTrue(await container.CheckIfExistsAsync(GalleryImageVersionName));
+            Assert.IsFalse(await container.CheckIfExistsAsync(GalleryImageVersionName + "1"));
+
+            Assert.ThrowsAsync<ArgumentNullException>(async () => _ = await container.CheckIfExistsAsync(null));
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task Get()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            var container = await GetGalleryImageVersionContainerAsync();
+            var vmContainer = _resourceGroup.GetVirtualMachines();
+            var vmName = Recording.GenerateAssetName("testVM-");
+            var vnet = await CreateVirtualNetwork();
+            var nic = await CreateNetworkInterface(GetSubnetId(vnet));
+            var vmInput = ResourceDataHelper.GetBasicLinuxVirtualMachineData(DefaultLocation, vmName, nic.Id);
+            var lroVm = await vmContainer.CreateOrUpdateAsync(vmName, vmInput);
+            _vm = lroVm.Value;
+            await _vm.DeallocateAsync();
+            await _vm.GeneralizeAsync();
+            var vmID = _vm.Id;
+            var BasicGalleryImageVersionData = ResourceDataHelper.GetBasicGalleryImageVersionData(DefaultLocation, vmID);
+            var GalleryImageVersionName = "1.0.0";
+            var lroImageVersion = await container.CreateOrUpdateAsync(GalleryImageVersionName, BasicGalleryImageVersionData);
+            GalleryImageVersion imageversion = lroImageVersion.Value;
+            GalleryImageVersion imageversion2 = await container.GetAsync(GalleryImageVersionName);
+
+            ResourceDataHelper.AssertGalleryImageVersion(imageversion.Data, imageversion2.Data);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task GetAll()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            var container = await GetGalleryImageVersionContainerAsync();
+            var vmContainer = _resourceGroup.GetVirtualMachines();
+            var vmName = Recording.GenerateAssetName("testVM-");
+            var vnet = await CreateVirtualNetwork();
+            var nic = await CreateNetworkInterface(GetSubnetId(vnet));
+            var vmInput = ResourceDataHelper.GetBasicLinuxVirtualMachineData(DefaultLocation, vmName, nic.Id);
+            var lroVm = await vmContainer.CreateOrUpdateAsync(vmName, vmInput);
+            _vm = lroVm.Value;
+            await _vm.DeallocateAsync();
+            await _vm.GeneralizeAsync();
+            var vmID = _vm.Id;
+            var BasicGalleryImageVersionData = ResourceDataHelper.GetBasicGalleryImageVersionData(DefaultLocation, vmID);
+            var GalleryImageVersionName1 = "1.0.0";
+            var GalleryImageVersionName2 = "1.0.0";
```

Comment:
We should not use same name for two different resources
```suggestion
            var GalleryImageVersionName1 = "1.0.0";
            var GalleryImageVersionName2 = "1.0.1";
```